tests/compile/main.js

```diff
@@ -30,6 +33,7 @@ goog.require('testBlocks');
 
 
 function init() {
+  Object.assign(Msg, window['Blockly']['Msg']);
```

Comment:
I note that `tests/compile/index.js` loads `msg/js/en.js`, which is the pre-wrapped version that does not yet have the `Object.assign` built in.

I presume the reason it wasn't working as it did previously (loading blockly as part of `main_compressed.js` and then loading `en.js`) is because `main_compressed.js` does not define a `Blockly.Msg` property any more due to advanced compilation.

Previously here was a hack to work around this in `core/blockly.ts`, but a while ago I moved it to `core/main.js`.  It _looks_ like `core/main.js` is still getting presented to Closure Compiler by `buildAdvancedCompilationTest`.  @NeilFraser: can you verify whether the code contained therein actually being included in the output?

* If it is then I'm not sure why it's not having the desired effect.  Can you see if `globalThis.Blockly` is unexpectedly defined by that point?

* If not, does putting something like:
  ```JS
  if (!('Blockly' in globalThis)) {
    globalThis['Blockly'] = {};
  }
  if (!('Msg' in globalThis['Blockly'])) {
    globalThis['Blockly']['Msg'] = 'Msg': Blockly.Msg;
  }
  ```
  in `tests/compile/main.js` resolve the loading issue without needing to resort to the `Object.assign` kludge?  Not that this one is much nicer, but at least it is working around a known issue with advanced compilation.